js_modules/dagit/packages/core/src/ui/SectionedLeftNav.tsx

```diff
@@ -0,0 +1,267 @@
+import {BaseTag, Box, Colors, Icon, IconWrapper, StyledTag} from '@dagster-io/ui';
+import * as React from 'react';
+import styled from 'styled-components/macro';
+
+import {AppContext} from '../app/AppContext';
+import {isAssetGroup} from '../asset-graph/Utils';
+import {useStateWithStorage} from '../hooks/useStateWithStorage';
+import {getJobItemsForOption, JobItem} from '../nav/FlatContentList';
+import {DagsterRepoOption, WorkspaceContext} from '../workspace/WorkspaceContext';
+import {buildRepoAddress} from '../workspace/buildRepoAddress';
+import {repoAddressAsString} from '../workspace/repoAddressAsString';
+import {RepoAddress} from '../workspace/types';
+
+const validateExpandedKeys = (parsed: unknown) => (Array.isArray(parsed) ? parsed : []);
+const EXPANDED_REPO_KEYS = 'dagit.expanded-repo-keys';
+
+export const SectionedLeftNav = () => {
+  const {loading, visibleRepos} = React.useContext(WorkspaceContext);
+  const {basePath} = React.useContext(AppContext);
+
+  const [expandedKeys, setExpandedKeys] = useStateWithStorage<string[]>(
+    basePath + ':' + EXPANDED_REPO_KEYS,
+    validateExpandedKeys,
+  );
+
+  const onToggle = React.useCallback(
+    (repoAddress: RepoAddress) => {
+      const key = repoAddressAsString(repoAddress);
+      setExpandedKeys((current) => {
+        let nextExpandedKeys = [...(current || [])];
+        if (nextExpandedKeys.includes(key)) {
+          nextExpandedKeys = nextExpandedKeys.filter((k) => k !== key);
+        } else {
+          nextExpandedKeys = [...nextExpandedKeys, key];
+        }
+        return nextExpandedKeys;
+      });
+    },
+    [setExpandedKeys],
+  );
+
+  const duplicateRepoNames = React.useMemo(() => {
+    const uniques = new Set<string>();
+    const duplicates = new Set<string>();
+    visibleRepos.forEach((repo) => {
+      const repoName = repo.repository.name;
+      if (uniques.has(repoName)) {
+        duplicates.add(repoName);
+      } else {
+        uniques.add(repoName);
+      }
+    });
+    return duplicates;
+  }, [visibleRepos]);
+
+  // Sort repositories alphabetically, then move empty repos to the bottom.
+  const sortedRepos = React.useMemo(() => {
+    const alphaSorted = [...visibleRepos].sort((a, b) =>
+      a.repository.name.toLocaleLowerCase().localeCompare(b.repository.name.toLocaleLowerCase()),
+    );
+    const reposWithJobs = [];
+    const reposWithoutJobs = [];
+    for (const repo of alphaSorted) {
+      const jobs = repo.repository.pipelines;
+      if (jobs.length > 1 && jobs.some((job) => !isAssetGroup(job.name))) {
```

Comment:
Hmm I think this should be `jobs.length > 0`, because it'd be possible to have a repo with no assets that doesn't expose an asset group job and only one standard job? i don't /think/ asset group jobs are generated for repos that have no assets.